src/core/unit.c

```diff
@@ -2225,6 +2225,11 @@ int unit_add_dependency(Unit *u, UnitDependency d, Unit *other, bool add_referen
                 return 0;
         }
 
+        if (d == UNIT_BEFORE && other->type == UNIT_DEVICE) {
+                log_unit_warning(u, "Dependency Before=%s ignored (.device units cannot be delayed)", other->id);
+                return 0;
+        }
+
```

Comment:
.device units appear the moment that udev reports them to be present. So they really cannot be delayed.

That said, I think it would be useful to allow Before dependencies on device units for transitive dependencies (A.unit has Before=B.device, C.unit has After=B.device, and that would mean that when shutting down, C would be stopped before A). Right now we cannot express such dependencies.